gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilPlugin.kt

```diff
@@ -388,23 +401,33 @@ internal class Variant private constructor(
       }
 
       val project = kotlinCompilation.target.project
-      val name = kotlinCompilation.name
+      val extension = project.extensions.getByType(AnvilExtension::class.java)
+      val androidVariant = (kotlinCompilation as? KotlinJvmAndroidCompilation)?.androidVariant
+
+      val commonFilter = CommonFilter(kotlinCompilation.identifier, extension)
+      val variantFilter = if (androidVariant != null) {
+        AndroidVariantFilter(commonFilter, androidVariant)
+      } else {
+        JvmVariantFilter(commonFilter)
+      }
 
       @Suppress("UNCHECKED_CAST")
       return Variant(
-        name = name,
+        name = kotlinCompilation.identifier,
         project = project,
-        extension = project.extensions.getByType(AnvilExtension::class.java),
         compileTaskProvider = kotlinCompilation.compileKotlinTaskProvider as
           TaskProvider<KotlinCompile>,
-        androidVariant = (kotlinCompilation as? KotlinJvmAndroidCompilation)?.androidVariant,
+        androidVariant = androidVariant,
         compilerPluginClasspathName = PLUGIN_CLASSPATH_CONFIGURATION_NAME +
           kotlinCompilation.target.targetName.capitalize(US) +
-          name.capitalize(US)
+          kotlinCompilation.name.capitalize(US),
+        variantFilter = variantFilter
       ).also {
         // Sanity check.
         check(it.compileTaskProvider.name.startsWith("compile"))
       }
     }
   }
 }
+
+private val KotlinCompilation<*>.identifier: String get() = name
```

Comment:
The Variant constructor takes in a parameter named "name", why'd you change it?

Reply:
Actually, CommonFilter too 🙈

Reply:
My thinking was to have a proxy for the getter. But it's not needed and I don't see a big benefit either anymore. Will change.